### DISCUSSION TITLE
Suport for Three-spool/Multi-spool turbofan/turboprop

### DISCUSSION THREAD

**Discussion by www2000** [JSBSim-Team/jsbsim#726] [Ideas]
The majority of Turbofan/Turbojet/Turboprop in today's aircraft having 2-spool and this reflect the current state of jsbsim.
In this idea is to upgrade jsbsim to support three-spool[1] and n-spool configs for Turbofan/Turbojet/Turboprop.
The default is 2-spool setup for backward compatibility the older config files when the `spools={n}` Attributes in the `turbine_engine` tag is missing in the XML config.
beside this the  `<spools n={number}> ... </spools>` make the XML code more readable. 

[1] https://en.wikipedia.org/wiki/Category:Three-spool_turbofan_engines

xml template with the change:
```xml
<?xml version="1.0"?> 
 <turbine_engine name="{string}" spools="{ number > 2}">
  <milthrust unit="{LBS | N}"> {number} </milthrust>
  <maxthrust unit="{LBS | N}"> {number} </maxthrust>
  <bypassratio> {number} </bypassratio>
  <bleed> {number} </bleed>
  <tsfc> {number} </tsfc>
  <atsfc> {number} </atsfc>
  <augmented> {0 | 1} </augmented>
  <augmethod> {0 | 1 | 2} </augmethod>
  <injected> {0 | 1} </injected>
  <injection-time> {number} </injection-time>
  <function name="IdleThrust"> </function>
  <function name="MilThrust">  </function>
  <function name="AugThrust">  </function>
  <function name="Injection">  </function>
  
  <spools n={number}>
    <ignitionn> {number} </ignitionn>
    <idlen> {number} </idlen>
    <spinup> {number} </spinup>
    <startrate> {number} </startrate>
    <max> {number} </max>
  </spools>
  
  <!-- alias for spools for backward compatibility (optional and will be ignore by detection of the <spool> tag) -->
  <!-- alias for n1 spool <spools n=1> ... </spools> -->
  <ignitionn1> {number} </ignitionn1>
  <idlen1> {number} </idlen1>
  <n1spinup> {number} </n1spinup>
  <n1startrate> {number} </n1startrate>
  <maxn1> {number} </maxn1>
  
  <!-- alias for n2 spool <spools n=2> ... </spools> -->
  <ignitionn2> {number} </ignitionn2>
  <idlen2> {number} </idlen2>
  <n2spinup> {number} </n2spinup>
  <n2startrate> {number} </n2startrate>
  <maxn2> {number} </maxn2>
  
 </turbine_engine>
```

**Comment by bcoconni** [comment]
Hi @www2000,

Just to avoid confusion: the turbine model in JSBSim is not an engineering model. No physics or thermodynamics laws are modeled: the spool rotation speeds `n1` and `n2` are simply faked based on a simple linear relationship between idling and the maximum value:

https://github.com/JSBSim-Team/jsbsim/blob/dafcaadc361ed1599d3d51b68c17dc8f93d699b4/src/models/propulsion/FGTurbine.cpp#L124-L129

The same goes for all the other engine parameters (EGT, oil temp, etc.). So handling a third spool could just be the matter of adding a new property `n3` to your XML model. This property would output whatever value you deem plausible and the realism would be just as good as if this third value was computed in the C++ code.

That being said, if you'd like to redesign the XML code for better legibility that's fine and you are welcome to submit a pull request. As long as they were properly written and did not break anything we have accepted all the pull requests that were adding features to the turbine model.
